apps/admin-x-activitypub/src/components/feed/ArticleModal.tsx

```diff
@@ -452,7 +455,11 @@ const ArticleModal: React.FC<ArticleModalProps> = ({
 
         // Update the replyCount on the current activity loaded in the modal
         // This is used for when we navigate via the history
-        object.replyCount = (object.replyCount ?? 0) + 1;
+        setReplyCount(replyCount + 1);
```

Comment:
I think it is preferable to pass a function when you are modifying state based on the current state - this can avoid things getting out of sync e.g.

```ts
const [state, setState] = useState(0);
setState(state + 1);
setState(state + 1);
setState(state + 1);
```

On the next render `state` will have a value of `1` NOT `3` - this is because the setStates are called synchronously and then the render happens after the batch

In this case I don't think it causes a bug, but I do think it's good practice to instead pass a function for increment/decrement like below

```ts
setReplyCount((count) => count + 1);
setReplyCount((count) => count - 1);
```

Reply:
Nice one, thanks for the nudge

Reply:
updated in https://github.com/TryGhost/Ghost/pull/22330/commits/4700cf870bd09a527abf1c32ad73cd22da227977